### ISSUE TITLE
Framework: No more site-picker pages

### ISSUE THREAD

**Issue by rralian** [Automattic/wp-calypso#7951]
While reviewing [another issue](https://github.com/Automattic/wp-calypso/issues/7812#issuecomment-245430537), it occurs to me that there's really no functional benefit to showing a "site-picker" page for sections in calypso when the URL does not contain a site. E.g.,

![plans site picker](https://cloudup.com/cRqtU_t6jmc+)

This is a rather awkward page, tbh. And the user needs to select a site in order to do anything useful. But why don't we just default to selecting the user's primary site? Or just redirect to the primary site? There's a good chance this is the site they want to work with. And if not, they just have to select a different site... just like they would have to do from the site picker page anyway.

/cc @mtias @artpi

**Comment by folletto** [comment]
> But why don't we just default to selecting the user's primary site? Or just redirect to the primary site? 

I agree that the page might get some more design iterations, but as is today is reusing the patterns of the site sidebar selector, so even if not ideal, creates a good resonance with the patterns we have.

The reason that page exists:

1. Every section should have a "All Sites" view. So in theory that page should never appear.
2. Many users will have just a single site, so redirect will happen naturally.
3. That page allows support URLs to be shared easily, as well as allowing cross-links between sections. A forced redirect for people with more than one site would always require them to look for the site switcher.
4. Forced redirect isn't an ideal pattern.
5. Even when not present on when a specific view isn't possible at all, it's very important that the  `/section` URL doesn't redirect, because it would break the semantic of the scheme.

In short, the enhancement here isn't much to rework this page, but to make sure we have "All Sites" pages available.

**Comment by rralian** [comment]
> as is today is reusing the patterns of the site sidebar selector, so even if not ideal, creates a good resonance with the patterns we have.

I think the site picker on this page is different enough from using the site picker in the sidebar that we're not really reinforcing a meaningful pattern.

> Many users will have just a single site, so redirect will happen naturally.

Right, it's not an issue for these users. Very few of our users are going to run into this, but it will happen for power users.

> Every section should have a "All Sites" view. So in theory that page should never appear.

Yes when it makes sense for a section to have an "all sites" view we should have one. But for better or worse, we use this "site picker" page in a number of places where an "all-sites" view either doesn't make sense or because we haven't had a chance to build it and this is our workaround.

> Forced redirect isn't an ideal pattern.

I don't feel strongly about doing this as a redirect. We could also just default to the primary site at the section level (when an "all sites" view doesn't make sense or hasn't yet been built).

> A forced redirect for people with more than one site would always require them to look for the site switcher.

I think this is your strongest point. If we think people are going to have difficulty finding the site-picker if we present them with their primary site when they land on a section-url (say, from a happiness-given link), then that's an issue. I do think people with multiple sites are more likely to be savvy and familiar with our interface though.

It would be interesting to know "of people who have multiple sites, how often do they interact with their primary site versus a non-primary site." I make this suggestion because I usually just want to do something on my primary site, so always having to select a site is a (minor) pain. If that's a really common case, I think it's worth considering this change. But on the other hand if multi-site users generally work across their sites, it's probably helpful to force them to pick a site before they start working with it.

**Comment by folletto** [comment]
> I make this suggestion because I usually just want to do something on my primary site, so always having to select a site is a (minor) pain. If that's a really common case, I think it's worth considering this change. But on the other hand if multi-site users generally work across their sites, it's probably helpful to force them to pick a site before they start working with it.

In abstract, I'd agree with this approach. In practice we always need to think in terms of flows, not screens: can you outline scenarios? How a user that opens wordpress.com and logs in would end up on one of our Site Select screens above?

**Comment by rralian** [comment]
It doesn't come up often. It usually has to do with needing a generic redirect landing page. In this case we are performing a redirect from the wordpress.com/pricing page to the calypso plans page when we detect the user is logged-in on wordpress.com/pricing (doing this to address the lack of an ability to upgrade your site from wordpress.com/pricing). The flow felt awkward to me, and my thought was "When would I want to see this site-picker page?" to which my honest answer is "Never." 😄

**Comment by rralian** [comment]
FWIW in that case #7812 I got this feedback from @supernovia in Happiness, which tells me we need a different solution for that issue, but I think it's applicable to this discussion as well.
> Our volunteers are complaining that they can't see pricing without selecting a site, and users who don't have sites can't access pricing at all

**Comment by supernovia** [comment]
Interesting discussion. I like the site picker when someone is going to take action on a site (adding an upgrade, deleting, etc.)  It's less helpful if someone is just looking for general information, as they have been with /pricing , though I can see how we may want to be able to show special upgrade pricing for a single site.

>  If we think people are going to have difficulty finding the site-picker if we present them with their primary site when they land on a section-url (say, from a happiness-given link), then that's an issue. I do think people with multiple sites are more likely to be savvy and familiar with our interface though.

You would think. But I have seen plenty of users inadvertently sign up for multiple sites, and those are the ones who would struggle most without a site picker.

I like the idea of multi-site views wherever we can implement them, but I understand that's a complex issue.  /settings for example wouldn't seem right in a multi site view, and HEs and volunteers give out that link all the time.

**Comment by folletto** [comment]
> Our volunteers are complaining that they can't see pricing without selecting a site, and users who don't have sites can't access pricing at all

+

>  It's less helpful if someone is just looking for general information, as they have been with /pricing, though I can see how we may want to be able to show special upgrade pricing for a single site.

Yep. This seems to me a problem in not having an All Sites page tho. It's not really solved by defaulting. Pricing doesn't seem to me a section that has any blocker for an All Sites view.

---

> I like the idea of multi-site views wherever we can implement them, but I understand that's a complex issue. /settings for example wouldn't seem right in a multi site view, and HEs and volunteers give out that link all the time.

If I understand correctly, this seems aligned to one of the reasons I outlined above why we have the picker: the flow there is correct in asking the user to pick a site first, then display the right settings.  :)
